app/assets/javascripts/discourse/widgets/widget.js.es6

```diff
@@ -270,18 +196,31 @@ export default class Widget {
     }
   }
 
-  attach(widgetName, attrs, opts) {
-    let WidgetClass = _registry[widgetName];
+  attach(widgetName, attrs, opts, otherOpts) {
+    const widgetNamesToTry = [widgetName];
```

Comment:
Rather than looping over an array of two, why not do `WidgetClass = _registry[widgetName] || _registry[opts.fallbackWidgetName]`? A lot less code.

Reply:
The current version looks up the widget name in the register. Is that actually necessary?

Reply:
I'm sorry, I wasn't clear at all. There are two registries. I'm investigating why looking up the widget name in `this.register` is necessary.

Reply:
I am not sure if that is used anymore. I suspect it is leftover code and can be removed. Worth looking into!

Reply:
There are several different widgets that don't use `createWidget` and so aren't registered in the widget registry. As things stand, this is necessary, but it wouldn't be difficult to fix. However, changing this might also break plugins.

Reply:
Can you give me an example of one that doesn't use createWidget? I forget how that API works.

Reply:
Have a look at `discourse/widgets/category-link.js.es6`, it's used in `discourse/widgets/search-menu-results.js.es6: 89 & 155`.

Reply:
Oh right that makes sense and would indeed break backwards compatibility.

What I'd suggest then is creating a helper method that returns the class or null. Then you can call that an extra time if a backup is provided.

We could also remove the register by having an initializer find all widgets and put them in the registry when the app boots (or perhaps the first time a widget is rendered.)